actions/api/mailchimp.py

```diff
@@ -44,6 +52,22 @@ def subscribe_user(self, list_id: Text, email: Text) -> bool:
             )
             return True
 
-        # if the user is already subscribed, return False
         except MailChimpError:
-            return False
+            try:
+                hash = self.hash_email(email)
+                status = self.client.lists.members.get(list_id, hash).get("status")
+                if status != "subscribed" and status != "pending":
+                    # if user is in database, but is unsubscribed or archived, attempt to resubscribe
+                    self.client.lists.members.update(
+                        list_id,
+                        hash,
+                        data={"email_address": email, "status": "pending"},
+                    )
+                    return True
+                return False
+            except MailChimpError:
+                return False
```

Comment:
We'll hit L67 when  user is subscribed or pending. We'll only hit L69 if something else goes wrong - e.g. MailChimp can't handle the email address, or fails to make an update because of a temporary interruption, something like that. 

Ja, I'm in favour of switching the order of conditions - we're already not relying on exceptions except for basically unknown mailchimp issues

Reply:
Actually wait! We'll also hit L69 when the user isn't in the database at all, because in that case retrieving the status will fail, so it will go to except. This seems faulty. Will look into it.

Reply:
Ok no, the first try block takes care of that. Added more explanatory comments. I think it makes sense to rely on try/excepts here, either first to try user creation or fetching user status, but we could further change the order of attmpts

Reply:
Sorry about that. Can you take a look at the updated function and how it's called in actions now?

Reply:
Yes, it looks great! The comments are super helpful. If it makes it easier we could always just return a string from `["user_subscribed", "already_subscribed", "failure"]` back to the actions to interpret -- but I think your comments are clear enough as well